### ISSUE TITLE
Transfer repositories

### ISSUE THREAD

**Issue by johndbritton** [education/classroom#438]
Just got a suggestion from a user in person that we should provide a way to transfer assignment repositories to the submitters in one step.

**Comment by hackerkid** [comment]
@johndbritton How do we decide the user to which the repo should be transfered in case of team projects?

**Comment by tarebyte** [comment]
> Maybe we could find a way to transfer it to all of them by duplicating it?

We since we have access to the importer API we can just duplicate and add it to their accounts on their behalf.

The problem I see is the public/private situation, do we keep repos private if the assignment was private. And what do we do if the student doesn't have any more free private repos.

**Comment by hackerkid** [comment]
@tarebyte @johndbritton Should the original repository be deleted once the import is complete?

**Comment by johndbritton** [comment]
> we can just duplicate and add it to their accounts on their behalf.

Only drawback here is we might not be able to recreate all the issues and pull requests.

> Should the original repository be deleted once the import is complete?

For the first pass, I would recommend just doing individual assignments and in the case of individual assignments, yes.

> Also should we give the ability to transfer all the repos in an assignment by a single click or one by one manually?

All in a single click for sure. Might be nice to have the ability to do it one by one, but not 100% sure that's needed. The use case I see here is that the course is over and the teacher wants to give the code to students rather than deleting it. Could be useful to do it for all assignments as one button for the whole classroom.

**Comment by sahilshekhawat** [comment]
Instead of transferring repository to the user, can we just fork it and sync it whenever it is updated??

**Comment by johndbritton** [comment]
> can we just fork it and sync it whenever it is updated?

It think that's out of scope for this feature. The goals are to clean up the org account when a course finishes and let students keep their code.

What would be the benefit to using forks and keeping them in sync?

**Comment by sahilshekhawat** [comment]
okay, I didn't understood it completely. But if it so, then we can fork it to respective students accounts and delete the main repositories in the organisation, right?

**Comment by sahilshekhawat** [comment]
Also, it would be easy if Github can add Transfer feature into its API itself.

**Comment by johndbritton** [comment]
If you do a fork, none of the issues will transfer over. I think moving the repository via transfer is the best solution.

**Comment by johndbritton** [comment]
> Also, it would be easy if Github can add Transfer feature into its API itself.

I agree, that would be ideal.

> Do you know if they plan to do something like this?

I don't know if that's part of their plans, but I will suggest it to them.

**Comment by kylekthompson** [comment]
Hey! I'd be interested in giving this a shot (assuming you still want help with it).

A couple questions first:

- I took a quick look at GitHub's transfer feature, and it looks like Issues/PRs/Projects aren't moved over. Is this something we are ok with? 
- Given that the developer plan now gives unlimited private repos, is there still a concern public/private situation? If I'm thinking about this the right way:
  - If it was public: no problem; keep it public.
  - If it was private: they (the students) either pay for private repos or have the education pack, so it should be safe to just keep it private.

**Comment by nwoodthorpe** [comment]
Over a year of inactivity, closing.
